src/backend/mailbox_manager/tests/api/mailboxes/test_api_mailboxes_update.py

```diff
@@ -0,0 +1,65 @@
+"""
+Unit tests for the mailbox API
+"""
+
+import json
+import re
+from logging import Logger
+from unittest import mock
+
+from django.test.utils import override_settings
+
+import pytest
+import responses
+from requests.exceptions import HTTPError
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from core import factories as core_factories
+
+from mailbox_manager import enums, factories, models
+from mailbox_manager.api.client import serializers
+from mailbox_manager.tests.fixtures.dimail import (
+    TOKEN_OK,
+    response_mailbox_created,
+)
+
+pytestmark = pytest.mark.django_db
+
+
+def test_api_mailboxes__update_anonymous_forbidden():
+    """Anonymous users should not be able to update a mailbox via the API."""
+    mailbox = factories.MailboxFactory()
+    saved_secondary = mailbox.secondary_email
+    import pdb
+
+    pdb.set_trace()
+    APIClient().get(
+        f"/api/v1.0/mail-domains/{mailbox.domain.slug}/mailboxes/{mailbox.pk}/"
+    )
+    response = APIClient().patch(
+        f"/api/v1.0/mail-domains/{mailbox.domain.slug}/mailboxes/{mailbox.pk}/",
+        {"secondary_email": "new_secondary@newdomain.fr"},
+    )
+    assert response.status_code == status.HTTP_401_UNAUTHORIZED
+
+    mailbox.refresh_from_db()
+    assert mailbox.secondary_email == saved_secondary
+
+
+def test_api_mailboxes__update_authenticated_failure():
+    """Authenticated but unauthoriezd users should not be able to update mailbox."""
+    client = APIClient()
+    client.force_login(core_factories.UserFactory())
+
+    mailbox = factories.MailboxFactory()
+    saved_secondary = mailbox.secondary_email
+    response = client.patch(
+        f"/api/v1.0/mail-domains/{mailbox.domain.slug}/mailboxes/{mailbox.pk}/",
+        {"secondary_email": "new_secondary@newdomain.fr"},
+        format="json",
+    )
+
+    assert response.status_code == status.HTTP_403_FORBIDDEN
```

Comment:
Returning a 403 indicates the resource exists but access is forbidden. If access if forbidden, I would be more comfortable with returning a 404, which does not even indicate the resource exist. 

In a way we could apply this logic to everything that is under a domain the user is not allowed to see. WDYT, @qbey @elvoisin  ?

Reply:
So, actually the 403 returned here is a 403 for the domain. It's the same 403 when the mailbox doesn't even exist. 

Should I modify the behavior for 404 to be returned when querying on a domain you're not supposed to know of ? This affects the mailboxes but also domains and all endpoints below